src/pydvl/valuation/samplers/base.py

```diff
@@ -8,7 +8,7 @@
 
 import logging
 from abc import ABC, abstractmethod
-from typing import Callable, Generic, TypeVar
+from typing import Callable, Generic, Optional, TypeVar
```

Comment:
Note that we import `annotations` from `__future__` to have the pipe operator for types (sort of)